src/toolboxes/spline/sobel.ts

```diff
@@ -0,0 +1,101 @@
+export function calculateSobel(imageData: ImageData): ImageData {
+  const width = imageData.width;
+  const height = imageData.height;
+
+  const kernelX = [
+    [-1, 0, 1],
+    [-2, 0, 2],
+    [-1, 0, 1],
+  ];
+
+  const kernelY = [
+    [-1, -2, -1],
+    [0, 0, 0],
+    [1, 2, 1],
+  ];
+
+  let sobelData: Uint8ClampedArray;
+  let greyscaleData: Uint8ClampedArray;
+
+  function bindPixelAt(data: Uint8ClampedArray) {
+    return function (x: number, y: number, channel?: number): number {
+      channel = channel || 0;
+      return data[(width * y + x) * 4 + channel];
+    };
+  }
+
+  function bindSetPixelAt(data: Uint8ClampedArray) {
+    return function (
+      x: number,
+      y: number,
+      r?: number,
+      g?: number,
+      b?: number,
+      alpha?: number
+    ): void {
+      r = r || 0;
+      g = g || 0;
+      b = b || 0;
+      alpha = alpha || 0;
```

Comment:
```suggestion
    return function (
      x: number,
      y: number,
      r = 0,
      g = 0,
      b = 0,
      alpha = 0
    ): void {
```

I think this is the same and you type less?